src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java

```diff
@@ -450,7 +454,9 @@ private void sendData(MPDbAdapter dbAdapter, MPDbAdapter.Table table, String[] u
                     } else {
                         logAboutMessageToMixpanel("Retrying this batch of events.");
                         if (!hasMessages(FLUSH_QUEUE)) {
-                            sendEmptyMessageDelayed(FLUSH_QUEUE, mFlushInterval);
+                            mBackOffTime = Math.min((long)Math.pow(2, mFailedRetries) * mFlushInterval, 30*60*1000); // max 30 min
+                            sendEmptyMessageDelayed(FLUSH_QUEUE, mBackOffTime);
+                            mFailedRetries++;
```

Comment:
Also, I think we want to limit the max value of `mFailedRetries` but not `mBackOffTime`. I don't think doing `mFailedRetries++` makes any sense if `(long)Math.pow(2, mFailedRetries) * mFlushInterval` is larger than `30 * 60 * 1000`.